src/components/NewSnippet.jsx

```diff
@@ -70,14 +70,14 @@ const NewSnippet = ({ dispatch, history, syntaxes }) => {
   const handleSyntax = syntax => ({ syntax })
   const handleContent = content => ({ content })
 
-  const getSyntaxes = () => {
+  const memoizedSyntaxes = useMemo(() => {
```

Comment:
What's the benefit of using `useMemo()`?

Reply:
`useMemo` will memorize result of passed function and will be re-rendered only if `syntaxes` has changed.